recipes-app/iot2050-efivarfs-helper/files/iot2050-efivarfs-helper.tmpl

```diff
@@ -0,0 +1,180 @@
+
+
+def dump_subprocess_error(e: subprocess.CalledProcessError):
+    err_print()
+    err_print("ERROR:\t" + str(e).strip())
+    stdout = str(e.stdout).strip()
+    stderr = str(e.stderr).strip()
+    if stdout != '':
+        err_print("ERROR:\tSTDOUT: " + stdout)
+    if stderr != '':
+        err_print("ERROR:\tSTDERR: " + stderr)
+
+
+def change_efivarfs_immutable_flag(immutable: bool):
+    if immutable:
+        action = 'Adding'
+        option = '+i'
+    else:
+        action = 'Removing'
+        option = '-i'
+
+    for file in os.listdir(EFIVARFS_MOUNT_POINT):
+        if (file.startswith('db-')
+                or file.startswith('KEK-')
+                or file.startswith('PK-')):
+            filepath = os.path.join(EFIVARFS_MOUNT_POINT, file)
+            print("{} the immutable flag of {}... ".format(action, filepath),
+                  end='')
+            cmd = 'chattr {} {}'.format(option, filepath)
+            result = subprocess.run(
+                cmd, shell=True, capture_output=True, check=True, text=True)
+            print("Done.")
+
+
+def do_prepare():
+    message_prepare_fail = "Fail to prepare the efivarfs for manipulating. \
+You may need to check the environment."
+
+    try:
+        mod_required = 'tee_stmm_efi'
+        print("Checking for the kernel module {}... ".format(mod_required),
+              end='')
+        cmd = 'lsmod'
+        result = subprocess.run(
+            cmd, shell=True, capture_output=True, check=True, text=True)
+        if len([i for i in result.stdout.split('\n')
+                if i.startswith("{} ".format(mod_required))]) == 1:
+            print("Done.")
+        else:
+            print("ERROR: {} is not loaded!".format(mod_required))
+            exit_on_failure(message_prepare_fail)
+    except subprocess.CalledProcessError as e:
+        dump_subprocess_error(e)
+        exit_on_failure(message_prepare_fail, code=e.returncode)
+
+    # tee-supplicant is used to serve the trusted applications running in
+    # optee-os for accessing RPMB partition.
+    try:
+        print("Checking for the service tee-supplicant... ", end='')
+        cmd = 'systemctl is-active tee-supplicant.service'
+        result = subprocess.run(
+            cmd, shell=True, capture_output=True, check=True, text=True)
+
+        print("Done.")
+    except subprocess.CalledProcessError as e:
+        print("ERROR: service is inactive!")
+        exit_on_failure(message_prepare_fail)
+
+    # efivarfs must be mounted with read & write option.
+    try:
+        print("Checking for efivarfs mounting... ", end='')
+        cmd = 'findmnt {}'.format(EFIVARFS_MOUNT_POINT)
+        result = subprocess.run(
+            cmd, shell=True, capture_output=True, check=True, text=True)
+        efivarfs_mount_error = 'not mounted.'
+        for line in result.stdout.split('\n'):
+            if line.startswith(EFIVARFS_MOUNT_POINT):
+                mount_options = line.strip().split(' ')[-1].split(',')
```

Comment:
... This can then be done with something along adapting the following lines:
```python
for item in (line.split() for line in open("/proc/mounts")
             if line.split()[1].startswith('/')):
    print(item)
```

Reply:
Exactly.